### ISSUE TITLE
Query :: allow projection of one entity from a nested collection without client-side evaluation

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#5546]
example:

```
ctx.Customers.Select(c => c.Orders.FirstOrDefault())
```

This issue is a superset of #3807 If we project a single column we can translate it into a simple subquery like so:

```
SELECT (
    SELECT TOP(1) [o].[CustomerID]
    FROM [Orders] AS [o]
    WHERE [c].[CustomerID] = [o].[CustomerID]
)
FROM [Customers] AS [c]
```

This doesn't work if more than one column is projected because scalar query is expected (one column, one row)

**Comment by AndriySvyryd** [comment]
@smitpatel This appears to be fixed now.

**Comment by smitpatel** [comment]
Better fix in #10001
